### ISSUE TITLE
Ability to update issue % completed and status when logging time spent

### ISSUE THREAD

**Issue by fgrehm** [fernandokosh/redmine_time_tracker#9]
Kind of a mix of "Update issue" and "Log time"

**Comment by delaitre** [comment]
Hmmm...I think this can be a good idea!
I have 2 remarks though:
- The time spent is not linear with the issue completed % in general. Thus, the completion could not be computed automatically.
- I think this should be done in Redmine core (and I think the first remark explains this one).

What do you think ?

**Comment by fgrehm** [comment]
Sorry I didn't make myself clear. For me, it would be enough just to show the fields so that I can update them, I don't expect the plugin to automatically calculate that for me.

Maybe we could just display the same view from "Update issue" with log time filled?

**Comment by delaitre** [comment]
Well, I totally agree with you. Showing the "update issue" page instead of the "log time" one is a lot more powerful. Indeed, I never noticed the time logging div in this page. You made my day !

I will change that as soon as possible.

(The automatic computation would be the cherry on the cake, but it's impossible...)

**Comment by fgrehm** [comment]
Cool, I was wondering that we could automatically set issue status to _In Progress_ when starting to track time if current status is _New_. What do you think?

**Comment by delaitre** [comment]
Please, open a new issue for that feature. It will be hard to follow otherwise.

**Comment by delaitre** [comment]
Well, I tried to implement this but it's a harder than expected.
The reason is the edit action for issue should be done using a POST request...
I don't now yet how to get it working (I've asked for some help in the redmine forum).

**Comment by fgrehm** [comment]
I've finally had some time to work on this and finally made it happen:
   http://github.com/fgrehm/redmine_time_tracker/tree/update_issue

It's not 100% tested but it can be a start

**Comment by delaitre** [comment]
I've tried to implement this too few days ago. As I don't really like (indeed, I really don't like) to copy/paste views I've submitted a little patch (1 line diff, see my redmine fork) to edavis10 which allows time_entry attributes to be specified in the params. Then, I just have to redirect to the issue show/edit view.
However, edavis10 has no time right now to review pull requests. I don't know when it will happen, and if my patch will be accepted.
Your implementation might be a temporary one if the patch is accepted.

**Comment by fgrehm** [comment]
I don't like my approach as well but it was the easiest way I could make it work. And as I always forget to update issue statuses after working on them, I'll have to stick with it :-)
Hopefully your patch will be accepted and we'll have a better solution in the future.
Cheers.

**Comment by delaitre** [comment]
I've opened the issue 5441 (http://www.redmine.org/issues/5441) on redmine's bug tracker with the patch.

**Comment by delaitre** [comment]
I've updated my patch but I will appreciate if someone can test it (i.e. running the functional unit test) as I have some trouble to get a proper redmine install right now.

**Comment by delaitre** [comment]
Ok, no more help needed, I fixed my patch which contained a bug and submit the new one to redmine dev team.

**Comment by delaitre** [comment]
Big news, my patch as been merged in Redmine trunk for Redmine 1.1 (http://www.redmine.org/projects/redmine/repository/revisions/4411). As soon as 1.1 is available (due in 43 days) I will implement this feature.

**Comment by delaitre** [comment]
This is available in 0.4.
